zebra-chain/src/block/hash.rs

```diff
@@ -20,7 +20,7 @@ use proptest_derive::Arbitrary;
 ///
 /// Note: Zebra displays transaction and block hashes in big-endian byte-order,
 /// following the u256 convention set by Bitcoin and zcashd.
-#[derive(Copy, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Serialize, Deserialize, Default)]
```

Comment:
We have avoided implementing `Default` on consensus-critical types, because it's easy to miss an incorrect use in a review. It's easy to hide a `default()` in a call like `unwrap_or_default()` or even more subtle methods.

Can we use an explicit method instead, or only activate the `Default` impl in tests using `cfg_attr`?

This also applies to `Header`, `Root`, `ValueCommitment`, `NotSmallOrderValueCommitment`, `CompactDifficulty`, and `Solution`. (The rest are ok in production.)

Reply:
Sure. Thanks for the heads-up.

Reply:
Done in [b0bdf17](https://github.com/ZcashFoundation/zebra/pull/8047/commits/b0bdf17009fcdae32e810e164e7a6595581db645).